### PR TITLE
Add ability to xmlquery derived attributes

### LINKED ISSUE #2351: Allow referencing case envs inside of shell commands
e.g.

```
$SHELL{echo $NUM_NODES * 6 | bc}
```

### PR DESCRIPTION
**Pull request by jgfouca**
Add ability to xmlquery derived attributes

Refactor xmlchange:
The code blocks that handled the two types of user input were diverging.
This commit encapsulates to core xmlchange operation. It also will cause
the user to be informed with a special error if they try to change a derived
attribute.

Test suite: code-checker, by-hand
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #2351 

User interface changes?: xmlquery will now accept derived attributes

Update gh-pages html (Y/N)?: N

Code review: @jedwards4b

### CODE REVIEW

File: scripts/Tools/xmlquery [hunk 1]
```diff
@@ -190,10 +190,14 @@ formatter_class=argparse.RawDescriptionHelpFormatter)
         args.type, args.valid_values, args.partial_match, args.file
 
 def get_value_as_string(case, var, attribute=None, resolved=False, subgroup=None):
-    thistype = case.get_type_info(var)
-    value = case.get_value(var, attribute=attribute, resolved=resolved, subgroup=subgroup)
-    if value is not None and thistype:
-        value = convert_to_string(value, thistype, var)
+    if var in ["THREAD_COUNT", "TOTAL_TASKS", "TASKS_PER_NODE", "NUM_NODES", "SPARE_NODES", "TASKS_PER_NUMA", "CORES_PER_TASK"]:
```
Comment (jedwards4b) [inline scripts/Tools/xmlquery]: I wonder if there should be a message attached to the output to the effect that these variables cannot be modified with xmlchange?

Comment (jgfouca) [reply scripts/Tools/xmlquery]: @jedwards4b , i'm not sure how to do that without cluttering the output in the case that the user wants the raw value.

Comment (jedwards4b) [reply scripts/Tools/xmlquery]: Can you use logger.info and suppress if --value is used?

Comment (jgfouca) [reply scripts/Tools/xmlquery]: @jedwards4b I have an idea.

**Comment by jgfouca** [comment]
@jedwards4b , see what you think now.

**Pull request merged**